crates/pathfinder/src/sync/headers.rs

```diff
@@ -249,38 +249,46 @@ impl VerifyHashAndSignature {
-                    transaction_commitment: header.transaction_commitment,
-                    transaction_count: header
-                        .transaction_count
-                        .try_into()
-                        .expect("ptr size is 64 bits"),
-                    event_commitment: header.event_commitment,
-                    event_count: header.event_count.try_into().expect("ptr size is 64 bits"),
-                    state_diff_commitment: header.state_diff_commitment,
-                    state_diff_length: header.state_diff_length,
-                    starknet_version: header.starknet_version,
-                    starknet_version_str: header.starknet_version.to_string(),
-                    eth_l1_gas_price: header.eth_l1_gas_price,
-                    strk_l1_gas_price: header.strk_l1_gas_price,
-                    eth_l1_data_gas_price: header.eth_l1_data_gas_price,
-                    strk_l1_data_gas_price: header.strk_l1_data_gas_price,
-                    receipt_commitment: header.receipt_commitment,
-                    l1_da_mode: header.l1_da_mode,
-                },
-                self.chain,
-                self.chain_id
-            ),
-            Ok(VerifyResult::Match(_))
-        )
+        let result = verify_block_hash(
+            BlockHeaderData {
+                hash: header.hash,
+                parent_hash: header.parent_hash,
+                number: header.number,
+                timestamp: header.timestamp,
+                sequencer_address: header.sequencer_address,
+                state_commitment: header.state_commitment,
+                transaction_commitment: header.transaction_commitment,
+                transaction_count: header
+                    .transaction_count
+                    .try_into()
+                    .expect("ptr size is 64 bits"),
+                event_commitment: header.event_commitment,
+                event_count: header.event_count.try_into().expect("ptr size is 64 bits"),
+                state_diff_commitment: header.state_diff_commitment,
+                state_diff_length: header.state_diff_length,
+                starknet_version: header.starknet_version,
+                starknet_version_str: header.starknet_version.to_string(),
+                eth_l1_gas_price: header.eth_l1_gas_price,
+                strk_l1_gas_price: header.strk_l1_gas_price,
+                eth_l1_data_gas_price: header.eth_l1_data_gas_price,
+                strk_l1_data_gas_price: header.strk_l1_data_gas_price,
+                receipt_commitment: header.receipt_commitment,
+                l1_da_mode: header.l1_da_mode,
+            },
+            self.chain,
+            self.chain_id
+        );
+        match result {
+            Ok(VerifyResult::Match(_)) => true,
+            Ok(VerifyResult::Mismatch) => {
+                tracing::debug!(block_number=%header.number, expected_block_hash=%header.hash, "Block hash mismatch");
```

Comment:
That's not part of `VerifyResult` unfortunately. But even if it were it's not much help in my experience. To diagnose a hash mismatch you always need the interim commitment values (that are used to compute the final hash) so that you can find the part where is a discrepancy.